src/battle_script_commands.c

```diff
@@ -15474,6 +15465,47 @@ static void Cmd_callnative(void)
 }
 
 // Callnative Funcs
+
+static void SaveBattlerTarget(u32 battler)
+{
+    gBattleStruct->savedBattlerTarget[gBattleStruct->savedTargetCount++] = battler;
```

Comment:
I would add some check that savedTargetCount doesn't overflow with the array count of savedTargets. Same with attacker